src/databricks/labs/ucx/install.py

```diff
@@ -971,6 +986,15 @@ def _get_cluster_policies_with_external_hive_metastores(self):
                     yield policy
                     break
 
+    def _check_policy_has_instance_pool(self, policy_id):
+        policy = self._ws.cluster_policies.get(policy_id=policy_id)
+        def_json = json.loads(policy.definition)
+        instance_pool = def_json.get("instance_pool_id")
+        if instance_pool is not None:
```

Comment:
```suggestion
        return instance_pool is not None
```
And if-else becomes redundant 😉